Packet++/header/SomeIpLayer.h

```diff
@@ -303,58 +303,174 @@ class SomeIpLayer : public Layer
 	 */
 	void setPayloadLength(uint32_t payloadLength);
 
-	/**
-	 * Get the Header Length With Payload, as is stated in the length field of the SOME/IP header.
-	 * @return uint32_t Length of the complete SOME/IP Pdu, including header and payload
-	 */
-	uint32_t getHeaderLengthWithPayload() const;
-
 	/**
 	 * @return A pointer for the layer payload, meaning the first byte after the header
 	 */
-	uint8_t *getPduPayload() const { return m_Data + sizeof(someiphdr);	}
+	virtual uint8_t *getPduPayload() const { return m_Data + getSomeIpHeaderLen(); }
 
 	/**
 	 * @return The size in bytes of the payload
 	 */
-	size_t getPduPayloadSize() const { return getHeaderLengthWithPayload() - sizeof(someiphdr); }
+	virtual size_t getPduPayloadSize() const { return getHeaderLen() - getSomeIpHeaderLen(); }
 
 	/**
 	 * Get the Length of the SOME/IP header inc payload
 	 * @return size_t
 	 */
-	size_t getHeaderLen() const;
+	virtual size_t getHeaderLen() const { return sizeof(uint32_t) * 2 + getLengthField(); }
 
 	/**
 	 * Does nothing for this layer
 	 */
-	void computeCalculateFields() {}
+	virtual void computeCalculateFields() {};
 
 	/**
-	 * Identifies next layer and tries to create it
+	 * Identifies the following next layers: SomeIpLayer, SomeIpTpLayer. Otherwise sets PayloadLayer
 	 */
 	void parseNextLayer();
 
 	/**
 	 * @return The string representation of the SOME/IP layer
 	 */
-	std::string toString() const;
+	virtual std::string toString() const;

```

Comment:
We don't override `toString()` in `SomeIpTpLayer` but we probably should, so `SomeIpLayer::toString()` only handles SOME/IP packets, and `SomeIpTpLayer::toString()` handles TP packets